benchmarks/benchmarks/ci/esmf_regridder.py

```diff
@@ -151,3 +158,213 @@ def time_lazy_regridding(self, cache):
+        src_data = da.ones([n_lats_src * n_lons_src, h], chunks=chunk_size)
+        src = Cube(src_data)
+        iris.save(src, file, chunksizes=chunk_size)
+        # Construct regridder with a loaded version of the grid for consistency.
+        loaded_src = iris.load_cube(file)
+        # While iris is not able to save meshes, we add these after loading.
+        mesh_coord_x, mesh_coord_y = mesh.to_MeshCoords("face")
+        loaded_src.add_aux_coord(mesh_coord_x, 0)
+        loaded_src.add_aux_coord(mesh_coord_y, 0)
+        regridder = MeshToGridESMFRegridder(loaded_src, tgt)
+
+        return regridder, file, mesh
+
+    def setup(self, cache):
+        regridder, file, mesh = cache
+        self.src = iris.load_cube(file)
+        mesh_coord_x, mesh_coord_y = mesh.to_MeshCoords("face")
+        self.src.add_aux_coord(mesh_coord_x, 0)
+        self.src.add_aux_coord(mesh_coord_y, 0)
+        cube = iris.load_cube(file)
+        cube.add_aux_coord(mesh_coord_x, 0)
+        cube.add_aux_coord(mesh_coord_y, 0)
+        self.result = regridder(cube)
+
+    def time_lazy_regridding(self, cache):
+        assert self.src.has_lazy_data()
+        regridder, _, _ = cache
+        _ = regridder(self.src)
+
+    def time_regridding_realisation(self, cache):
+        assert self.result.has_lazy_data()
+        _ = self.result.data
+
+
+class TimeGridToMeshRegridding:
+    params = ["similar", "large source", "large target", "mixed"]
+    param_names = ["source/target difference"]
+
+    def setup(self, type):
+        lon_bounds = (-180, 180)
+        lat_bounds = (-90, 90)
+        n_lons_src = 20
+        n_lats_src = 40
+        n_lons_tgt = 20
+        n_lats_tgt = 40
+        h = 100
+        if type == "large source":
+            n_lons_src = 100
+            n_lats_src = 200
+        if type == "large target":
+            n_lons_tgt = 100
+            n_lats_tgt = 200
+        if type == "mixed":
+            coord_system_src = RotatedGeogCS(0, 90, 90)
+        else:
+            coord_system_src = None
+        grid = _grid_cube(
+            n_lons_src,
+            n_lats_src,
+            lon_bounds,
+            lat_bounds,
+            coord_system=coord_system_src,
+        )
+        src_data = np.arange(n_lats_src * n_lons_src * h).reshape(
+            [n_lats_src, n_lons_src, h]
+        )
+        src = Cube(src_data)
+        src.add_dim_coord(grid.coord("latitude"), 0)
+        src.add_dim_coord(grid.coord("longitude"), 1)
+        tgt_data = np.zeros(n_lats_tgt * n_lons_tgt)
+        tgt = Cube(tgt_data)
+        mesh = _gridlike_mesh(n_lons_tgt, n_lats_tgt)
+        mesh_coord_x, mesh_coord_y = mesh.to_MeshCoords("face")
+        tgt.add_aux_coord(mesh_coord_x, 0)
+        tgt.add_aux_coord(mesh_coord_y, 0)
+        self.regridder = GridToMeshESMFRegridder(src, tgt)
+        self.src = src
+
+    def time_perform_regridding(self, type):
+        _ = self.regridder(self.src)
+
+
+class TimeLazyGridToMeshRegridding:
```

Comment:
Could you write things in such a way that it inherits from `TimeLazyMeshToGridRegridding` but with just a few things swapped round?

Reply:
I don't think these classes would work that well together, mainly because the `setup_cache` in `TimeLazyMeshToGridRegridding` additionally returns a `mesh` argument which is required in `setup`. I think this would cause confusion in the method signatures.